zebrad/src/commands/start/sync.rs

```diff
@@ -389,26 +437,38 @@ where
 
             tracing::debug!(?hash, "requested block");
 
-            let span = tracing::info_span!("block_fetch_verify", ?hash);
+            // This span is used to help diagnose sync warnings
+            let span = tracing::warn_span!("block_fetch_verify", ?hash);
             let mut verifier = self.verifier.clone();
-            let task = tokio::spawn(
-                async move {
+            let task = tokio::spawn(async move {
+                async {
```

Comment:
I tried pulling this down locally and I think we can get rid of the double async block and save a little on indentation, tho it probably optimizes to the same code so not a big deal either way.

Github won't let me make a suggestion this big tho so here's the code, you'll need to also bring `TryFutureExt` into scope.

```rust
            let task = tokio::spawn(
                async move {
                    let block = match block_req.await {
                        Ok(zn::Response::Blocks(blocks)) => blocks
                            .into_iter()
                            .next()
                            .expect("successful response has the block in it"),
                        Ok(_) => unreachable!("wrong response to block request"),
                        // Make sure we can distinguish download and verify timeouts
                        Err(e) => Err(eyre!(e)).wrap_err("failed to download block")?,
                    };
                    metrics::counter!("sync.downloaded.block.count", 1);

                    let result = verifier
                        .ready_and()
                        .await
                        .map_err(|e| eyre!(e))
                        .wrap_err("verifier service failed to be ready")?
                        .call(block)
                        .await
                        .map_err(|e| eyre!(e))
                        .wrap_err("failed to verify block")?;
                    metrics::counter!("sync.verified.block.count", 1);
                    Result::<block::Hash, Report>::Ok(result)
                }
                .instrument(span)
                .map_err(move |e| (e, hash)),
            );
```

Reply:
I'll try it and see how it goes.

Feel free to create big changes in another PR or a maintainer commit next time, if you'd like.